src/Clock.tsx

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable no-console */
+import React, { useEffect, useState, useMemo } from 'react';
+
+type Props = {
+  name: string;
+};
+
+export const Clock: React.FC<Props> = ({ name }) => {
+  const [clockState, setClockState] = useState('');
+
+  let timerId = 0;
+
+  useEffect(() => {
+    timerId = window.setInterval(() => {
+      const date = new Date();
+
+      setClockState(date.toLocaleTimeString());
+    }, 1000);
+
+    return () => clearInterval(timerId);
+  }, [clockState]);
+
+  useMemo(() => {
+    console.info(clockState);
+  }, [clockState]);
```

Comment:
Same incorrect use of useMemo here. I guess it's better to use useEffect for this purpose